spec/type_references_and_ownership.md

```diff
@@ -0,0 +1,503 @@
+- BatchAccount -> Pool ([json schema](https://schema.management.azure.com/schemas/2017-09-01/Microsoft.Batch.json))
+- ResourceGroup -> Any resource
+
+This relationship tells us two things:
+- Where to create/manage the dependent resource (this `Route` goes in this particular `RouteTable`, this `RouteTable` has this `Route`)
+- That the dependent resource should be deleted when the parent resource is deleted. TODO: Race conditions here
+    - There are theoretically two cases here:
+        - The dependent resource must be deleted before the parent can be deleted.
+        - Deletion of the parent automatically cascades to all dependent resources. To the best of my knowledge all
+         owning resource relationships in Azure are this kind.
+
+Note that sometimes an owning type has its dependent type embedded as a property 
+(for example: `RouteTable` has the property [RouteTablePropertiesFormat](https://schema.management.azure.com/schemas/2020-03-01/Microsoft.Network.json)).
+Most types do not embed the dependent resource directly in the owning resource. We need to cater to both scenarios.
+
+## What do these relationships look like in existing solutions?
+
+This section examines how other operator solutions have tackled these problems. We look at:
+- ARM templates
+- AzureServiceOperator (ASO)
+- k8s-infra
+
+### Related/Linked resources
+
+#### What ARM does
+These are just properties (often but not always called `id`) which refer to the fully qualified ARM ID of another resource. For example see a
+[sample deployment template for a VMSS refering to an existing vnet](https://github.com/Azure/azure-quickstart-templates/blob/master/201-vmss-existing-vnet/azuredeploy.json#L136).
+
+```json
+"properties": {
+  "subnet": {
+    "id": "[resourceId(parameters('existingVnetResourceGroupName'), 'Microsoft.Network/virtualNetworks/subnets', parameters('existingVnetName'), parameters('existingSubNetName'))]"
+  },
+}
+```
+
+#### What ASO does
+Similar to what ARM templates do, ASO uses the decomposition of fully qualified resource id to reference another resource, as seen [here for VMSS -> VNet](https://github.com/Azure/azure-service-operator/blob/92240406aff3863f3a267d8a1dc1e28aa3e841ae/api/v1alpha1/azurevmscaleset_types.go#L25)
+
+```go
+type AzureVMScaleSetSpec struct {
+    ...
+	Location               string `json:"location"`
+	ResourceGroup          string `json:"resourceGroup"`
+	VirtualNetworkName     string `json:"virtualNetworkName"`
+	SubnetName             string `json:"subnetName"`
+    ...
+}
+```
+
+These properties are combined into a fully qualified ARM ID like so:
+
+```go
+subnetIDInput := helpers.MakeResourceID(
+    client.SubscriptionID,
+    resourceGroupName,
+    "Microsoft.Network",
+    "virtualNetworks",
+    vnetName,
+    "subnets",
+    subnetName,
+)
+```
```

Comment:
Perhaps extend the example to show what the result looks like, for those not so familiar with ARM.